### PR TITLE
refactor getClibInfo and remove @trusted from the related code (it's all @safe)

### PR DESCRIPTION
**Pull request by braddr**

### CODE REVIEW

File: compiler/src/dmd/backend/dtype.d [hunk 1]
```diff
@@ -40,12 +40,14 @@ private __gshared
     int type_num,type_max;   /* gather statistics on # of types      */
 }
 
+type* tsclib;
```
Comment (thewilsonator) [inline compiler/src/dmd/backend/dtype.d:43]: I don't think it is desired to make these thread local

File: compiler/src/dmd/backend/x86/cod1.d [hunk 2]
```diff
@@ -2069,12 +2069,11 @@ struct ClibInfo
     byte pop87;                         // # of pops off of the 8087 stack
 }
 
-__gshared int clib_inited = false;          // true if initialized
+int clib_inited = false;          // true if initialized
```
Comment (thewilsonator) [inline compiler/src/dmd/backend/x86/cod1.d:2072]: ditto

File: compiler/src/dmd/backend/dtype.d (hunk 1 above)
Comment (braddr) [reply compiler/src/dmd/backend/dtype.d:43]: I had a similar concern, but my thinking was that I _think_ that not gshared is an improvement, though a minor one.  In practice, none of this code is threaded or has any likelihood of being thread safe any time soon.

**Comment by braddr** [comment]
looks like one of the CI's is busted -- it's failing, I believe, before it's gotten near my changes:

Run dmd -run dmd/compiler/src/build.d -j$N MODEL=64
No DMD compiler is installed. Try AUTO_BOOTSTRAP=1 or manually set the D host compiler with HOST_DMD
DETAILS:

Error: Process completed with exit code 1.

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**